### ISSUE TITLE
Dynamic Resizing of histograms.

### ISSUE THREAD

**Issue by jason-neal** [DanielAndreasen/SWEETer-Cat#25]
Currently, the histogram size is set relative to the initial main figure size. 

If a user wants to make the plots bigger using the "Resize" tool then the histograms do not automatically scale in relation. This looks odd and then the histograms don't line up with the values. 

Need some sort of dynamic scaling to keep the shared axis the same size as the main plot.

Or just disable the ability to "Resize" the figure.

**Comment by DanielAndreasen** [comment]
In line with this (could be a new issue):
When selecting a sub-sample from the scatter plot, it would be nice if this is highlighted in the histogram.

**Comment by jason-neal** [comment]
Do you mean similar to the highlighting [here](https://demo.bokehplots.com/apps/selection_histogram). [Source code](https://github.com/bokeh/bokeh/blob/master/examples/app/selection_histogram.py) (copied from #20). 
As mentioned there it might be that we need to run a bokeh server, that might enable the dynamic scaling also.

Or we may just need to find the right part of bokeh to use.

**Comment by DanielAndreasen** [comment]
Yes, exactly like that. I will have a look at some point to see how this can be implemented with flask.

**Comment by DanielAndreasen** [comment]
Looks like there is some examples [here](http://bokeh.pydata.org/en/latest/docs/user_guide/server.html#embedding-bokeh-server-as-a-library) ([source code for Flask](https://github.com/bokeh/bokeh/blob/0.12.6/examples/howto/server_embed/flask_embed.py)).

**Comment by jason-neal** [comment]
This issue may disappear if we move to mpld3. (the sub-sample selection anyway)
